src/main/kotlin/io/newm/kogmios/protocols/messages/MsgEvaluateTxResponse.kt

```diff
@@ -0,0 +1,230 @@
+package io.newm.kogmios.protocols.messages
+
+import io.newm.kogmios.ClientImpl.Companion.json
+import io.newm.kogmios.protocols.model.ExecutionUnits
+import io.newm.kogmios.protocols.model.UtxoInput
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+/**
+ * Response that comes back from Ogmios after a MsgEvaluateTx message is sent.
+ */
+@Serializable
+@SerialName("EvaluateTx")
+data class MsgEvaluateTxResponse(
+    @SerialName("result")
+    val result: EvaluateTxResult,
+    @SerialName("reflection")
+    override val reflection: String,
+) : JsonWspResponse()
+
+@Serializable(with = EvaluateTxResultSerializer::class)
+sealed class EvaluateTxResult
+
+@Serializable(with = EvaluationResultSerializer::class)
+class EvaluationResult : EvaluateTxResult(), MutableMap<String, ExecutionUnits> by mutableMapOf() {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is EvaluationResult) return false
+        return true
```

Comment:
Could be simplified to:
```
return other is EvaluationResult
```

Reply:
This is generated code so I'm going to assume it's already optimal enough.